arkouda/categorical.py

```diff
@@ -189,6 +191,29 @@ def from_codes(
             **kwargs,
         )
 
+    @classmethod
+    def from_return_msg(cls, rep_msg) -> Categorical:
+        """
+        Create categorical from return message from server
+
+        Notes
+        ------
+        This is currently only used when reading a Categorical from HDF5 files.
+        """
+        # parse return json
+        eles = json.loads(rep_msg)
+        codes = create_pdarray(eles["codes"])
+        cats = Strings.from_return_msg(eles["categories"])
+        na_code = create_pdarray(eles["_akNAcode"])
+
+        segments = None
+        perm = None
+        if "segments" in eles and "permutation" in eles:
+            segments = create_pdarray(eles["segments"])
+            perm = create_pdarray(eles["permutation"])
```

Comment:
Based on the `init` method these two would be present or not. They are both coming from the GroupBy, so I think this needs to remain as is because we should have both or neither.